apps/fortuna/src/keeper.rs

```diff
@@ -188,28 +194,44 @@ pub async fn process_event(
 
                 if gas_estimate > gas_limit {
                     tracing::error!(
-                        "Gas estimate for reveal with callback is higher than the gas limit for chain: {}",
+                        "Chain: {} - gas estimate for reveal with callback is higher than the gas limit",
                         &chain_config.id
                     );
                     return Ok(());
                 }
 
-                let res = contract
+                let contract_call = contract
                     .reveal_with_callback(
                         event.provider_address,
                         event.sequence_number,
                         event.user_random_number,
                         provider_revelation,
                     )
-                    .gas(gas_estimate)
-                    .send()
-                    .await?
-                    .await;
+                    .gas(gas_estimate);
+
+                let res = contract_call.send().await;
+
+                let pending_tx = match res {
+                    Ok(pending_tx) => pending_tx,
+                    Err(e) => match e {
+                        ContractError::ProviderError { e } => return Err(e.into()),
+                        _ => {
```

Comment:
Please add some comment on what sort of other errors can happen here and why it is ok to ignore them and return Ok(())